samples/create_deployment.js

```diff
@@ -0,0 +1,59 @@
+// Copyright 2020, Google LLC.
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+'use strict';
+
+/**
+ * Create a Game Servers Deployment.
+ * @param {string} projectId string project identifier
+ * @param {string} deploymentId unique identifier for the new Game Server Deployment
+ */
+function main(
+  projectId = 'YOUR_PROJECT_ID',
+  deploymentId = 'DEPLOYMENT_ID'
+) {
+  // [START cloud_game_servers_create_deployment]
+  /**
+   * TODO(developer): Uncomment these variables before running the sample.
+   */
+  // const projectId = 'Your Google Cloud Project ID';
+  // const deploymentId = 'A unique ID for the Game Server Deployment';
+  const {
+    GameServerDeploymentsServiceClient,
+  } = require('@google-cloud/game-servers');
+
+  const client = new GameServerDeploymentsServiceClient();
+
+  async function createGameServerDeployment() {
+    const request = {
+        parent: `projects/${projectId}/locations/global`,
+        deploymentId: deploymentId,
+        gameServerDeployment: {
+            description: "nodejs test deployment",
+        },
+    };
+
+    const [operation] = await client.createGameServerDeployment(request);
+    const [result] = await operation.promise();
+
+    console.log('Game Server Deployment created:');
+    console.log(`\t Deployment name: ${result.name}`);
+    console.log(`\t Deployment description: ${result.description}`);
+    
+    // [END cloud_game_servers_create_deployment]
+  }
+
+  createGameServerDeployment();
+}
+
+main(...process.argv.slice(2));
```

Comment:
The guidance here kinda annoyingly changed recently.  We want to add this to the bottom of each sample:
```js
main(...process.argv.slice(2)).catch(err => {
  console.error(err.message);
  process.exitCode = 1;
});
process.on('unhandledRejection', (err) => {
  console.error(err.message);
  process.exitCode = 1;
});
```

Reply:
I've been having problem with this. It is failing with error that "Cannot read property 'catch' of undefined.

Reply:
That will happen if your `main` function is not `async`, which it needs to be.